File: 백준/Gold/14921. 용액 합성하기/용액 합성하기.java
import java.util.*;
import java.io.*;

public class Main {
    static int n, answer = Integer.MAX_VALUE;
    static int[] A;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        n = Integer.parseInt(br.readLine());
        A = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) A[i] = Integer.parseInt(st.nextToken());

        int left = 0;
        int right = n-1;
        while (left < right) {
            int sum = A[left] + A[right];

            if (sum == 0) {
                answer = 0;
                break;
            }

            if (Math.abs(answer) > Math.abs(sum)) answer = sum;

            if (sum < 0) left++;
            else right--;
        }

        bw.write(String.valueOf(answer));
        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Gold/14921. 용액 합성하기/용액 합성하기.cc
#include <iostream>
#include <vector>
#include <climits>

using namespace std;

int answer = INT_MAX;
int n;
vector<int> A;

int main() {
	ios_base::sync_with_stdio(false);cout.tie(NULL);cin.tie(NULL);
	cin >> n;
	A.resize(n);
	for (int i = 0; i < n; i++) cin >> A[i];

	int left = 0;
	int right = n-1;

	while (left < right) {
		int sum = A[left] + A[right];

		if (sum == 0) {
			cout << 0;
			return 0;
		}

		if (abs(answer) > abs(sum)) answer = sum;

		if (sum < 0) left++;
		else right--;
	}

	cout << answer;
}